bundles/SimpleBackendSearchBundle/src/Model/Search/Backend/Data.php

```diff
@@ -95,9 +95,9 @@ class Data extends AbstractModel
     /**
      * User-ID of the owner
      *
-     * @var int
+     * @var int|null
      */
-    protected int $userOwner;
+    protected ?int $userOwner = null;
```

Comment:
should be the setter also `null`?
![image](https://github.com/pimcore/pimcore/assets/6014195/0d378456-75f1-457e-a0dc-88dd34f58c48)
![image](https://github.com/pimcore/pimcore/assets/6014195/4360793e-b274-42d4-8aeb-ff9ce59b3901)
https://github.com/pimcore/pimcore/blob/d982fa8d96dc775e49a04017cfa657882cedf148/bundles/SimpleBackendSearchBundle/src/Model/Search/Backend/Data/Dao.php#L83

Reply:
I did have a look into other setters for nullable fields, for example `setUserModification`, `setData` and `setCreationDate` etc. before issueing this PR. 
Did not change the `setUserOwner` signature to keep it consistent with the others.
The code you referenced is not calling any setters.

With the code in this MR the search indexer works fine for 1.5 million objects and assets.
Except that we get an unserialisation error on an asset crashing the command. Will have a look into that next week.

It makes sense to me to allow a field with a value to be set to `null`, thus changing the signature of the setters. This is an architectonic question I suppose. I suggest to do that in other PR, and maybe also look into the other data objects if they have similar issues.

Reply:
Hmm having a closer look at your comment, I see you were able to reproduce issue on quicksearch with the setter called with `null`.
I will change the signature. Do I also have to change the others to be consistent?

Reply:
I have changed setters of `modificationDate`, `userModification` and `userOwner`

Reply:
@bramstroker @kingjia90 Should we also change the setter of `creationDate`?

Reply:
@bramstroker Thank you for the changes!
@blankse probably yes, i've took a quick look, overall all the `setCreationDate()` in models are not accepting `null` but the getter usually do. The `search_backend_data` DAO due its nature and usage, it is "relying/tied" on these nullable getters.

Reply:
Alternatively could it be fixed by null coalescing all the values in these setters
https://github.com/pimcore/pimcore/blob/91a8f819595f06d00758033d2e066357ca33635c/bundles/SimpleBackendSearchBundle/src/Model/Search/Backend/Data/Listing/Dao.php#L50-L61

Reply:
Looks like the suggested nullable setter example here was wrong, i changed it manually on the DB, but wasn't aware of the *magic* in the screenshot below, the value would be always set and when not, would be 0.
https://user-images.githubusercontent.com/6014195/267583592-9dc926e5-c697-4a2f-b2da-96581e23c839.png

The fact that it crashes on non-nullable property is something that we are facing recently with the unserialized elements, so could be just that case.

So in short, only the changes in the first commit were correct. @bramstroker  Could you please revert the setter change?

Reply:
@kingjia90 I have removed last two commits by force push.
Should be ok now.

I don't have time to deep dive into this.

But one thing to consider is `0` is the system user id. Which is correct when the system user actually created or modified the object.
`null` however is another case, because that's the scenario when a user is removed from the system, therefore we cannot know the user anymore. Setting it to 0 is imo just wrong, because than it would be listed as system user in the frontend, which might cause confusion.
Not a really big deal I think, but it's something to consider.

I suggest to merge this PR as it at least solves the search reindex for now, which is currently broken for our installation and probably also for others using Pimcore 11.
Then you pimcore core devs can decide how to solve this case throughout pimcore codebase and take next steps.

Reply:
Thank you, the point of deleted users is something that i didn't take in account at all, i would take a deeper look into it as well.

Reply:
merging this Pr to the moment and would open a new issue to discuss about this null by deletion